packages/ui/src/theme/createTheme.ts

```diff
@@ -74,6 +82,53 @@ export function createTheme(
   theme?: Theme,
   baseTheme: BaseTheme = defaultTheme
 ): WebTheme {
+  /*
+1. find path to deprecated values in baseTheme
+2. see if values exist in theme that match depricated paths from base
+3. determine undepricated path
+4. see if values in theme exist for undepricated paths
+5. if no value of undepricated exists then replace it with value of depricated
+ */
+  if (theme) {
+    let deprecated: pathObject[] = findDeprecatedStateTokens(baseTheme);
+
+    deprecated.forEach((element: pathObject) => {
+      //look for a deprecated value in the theme
+      const path: string[] = element.path;
+      const deprecatedThemeValue = pathToValue(theme, path);
+
+      //use the deprecated path to generate the new path by replacing any state names with _{state}
+      const targetPath = path.map((pathValue) => {
+        if (STATE_NAMES.includes(pathValue)) {
+          return `_${pathValue}`;
+        }
+        return pathValue;
+      });
+
+      //follow the path to determine if a value already exists within the theme for the non deprecated version
+      let themeValue = theme.tokens || {};
+      targetPath.every((pathValue) => {
+        if (themeValue[pathValue]) {
+          themeValue = themeValue[pathValue];
+          return true;
+        } else {
+          themeValue = undefined;
```

Comment:
Is this step necessary?

Reply:
This step is here because we are mutating themeValue as we iterate through the loop and expecting themeValue to either be the path that was found or undefined at the end.  This same operation is more commonly done using recursive functions where you pass in the sub object to the next recursive function call instead of mutating a value as your crawl through the object, but in this case I felt the loop was easier to read and that overall the algorithm was easier to follow keeping this logic in place instead of breaking it out into a recursive function call, but I can switch this to a recursive function call if you want.